librz/core/cmd/cmd_print.c

```diff
@@ -7288,4 +7224,44 @@ RZ_IPI RzCmdStatus rz_cmd_base64_decode_handler(RzCore *core, int argc, const ch
 	rz_cons_println((const char *)buf);
 	free(buf);
 	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_bitstream_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	int len = (int)rz_num_math(core->num, argv[1]);
+	int skip = (int)rz_num_math(core->num, argv[2]);
+	if (len < 0 || skip < 0) {
+		RZ_LOG_ERROR("len and skip should be positive numbers\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	// `pb len skip` means skip <skip> bits then print <len> bits
+	char *buf = RZ_NEWS0(char, len + skip + 1);
+	if (!buf) {
+		RZ_LOG_ERROR("Fail to allocate memory\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_str_bits(buf, core->block, len + skip, NULL);
+	rz_cons_println(buf + skip);
+	free(buf);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_print_byte_bitstream_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	ut64 cur_off = core->offset;
+	int len = (int)rz_num_math(core->num, argv[1]);
+	if (len < 0) {
+		len *= -1;
+		rz_core_seek(core, cur_off - len, SEEK_SET);
+		rz_core_block_read(core);
```

Comment:
just use `rz_io_read_at` instead of moving the seek.

create a buffer with RZ_NEWS, use it to store things read from `rz_io_read_at`, then pass it to rz_str_bits instead of core->block.

Reply:
This way you don't need to change the core->seek and restore it back at the end of the function.